benchmarks/decomposition.py

```diff
@@ -34,6 +39,57 @@ def peakmem_fit(self, *args):
 
 class DictionaryLearning_bench(Benchmark):
     """
-    Benchmarks for DictionnaryLearning.
+    Benchmarks for DictionaryLearning.
     """
-    pass
+
+    # params = (fit_algorithm)
+    param_names = ["params"] + Benchmark.param_names
+    params = ([("lars",), ("cd",)],) + Benchmark.params
+
+    def setup(self, params, *common):
+        self.data = _decomposition_dataset()
+        fit_algorithm = params[0]
+        n_jobs = common[0]
+        self.dl_params = {
+            "n_components": 15,
+            "fit_algorithm": fit_algorithm,
+            "alpha": 0.1,
+            "tol": 1e-16,
+            "random_state": 42,
+            "n_jobs": n_jobs,
+        }
+
+    def time_fit(self, params):
+        estimator = DictionaryLearning(**self.dl_params)
+        estimator.fit(self.data)
+
+    def peakmem_fit(self, params):
+        estimator = DictionaryLearning(**self.dl_params)
+        estimator.fit(self.data)
```

Comment:
why is there an extra function for peakmem? Is there an example where we need to do something different than when getting the time? Looks unclean conceptually and engineering wise. A single function which is wrapped with the right measurement seems cleaner to me.

Reply:
This is actually one of `asv` 's benchmark type [link](https://asv.readthedocs.io/en/stable/writing_benchmarks.html#peak-memory) and it was requested by Jeremy in the previous PR.  I agree that it looks unclean but that's how asv works.